libphobos/m4/druntime/libraries.m4

```diff
@@ -30,3 +30,27 @@ AC_DEFUN([DRUNTIME_LIBRARIES_ZLIB],
     [system_zlib=yes],[system_zlib=no])
   AM_CONDITIONAL([DRUNTIME_ZLIB_SYSTEM], [test "$system_zlib" = yes])
 ])
+
+# DRUNTIME_LIBRARIES_ATOMIC
+# -------------------------
+# Allow specifying whether to use libatomic for
+# 128 bit atomic support.
+AC_DEFUN([DRUNTIME_LIBRARIES_ATOMIC],
+[
+  AC_ARG_WITH(libatomic,
+    AS_HELP_STRING([--without-libatomic],
+                   [Do not use libatomic in core.atomic (default: auto)]))
+
+  DCFG_HAVE_LIBATOMIC=false
+  AS_IF([test "x$with_libatomic" != "xno"], [
+    AC_CHECK_LIB([atomic], [__atomic_load_4], [
+        LIBS="-latomic  $LIBS"
```

Comment:
Is the link check strictly necessary?  It is a convenience library provided by gcc afterall.